commands/src/main/java/org/wildfly/extras/creaper/commands/security/realms/GroupToPrincipal.java

```diff
@@ -0,0 +1,146 @@
+package org.wildfly.extras.creaper.commands.security.realms;
+
+/**
+ * Helper class which holds information group-to-principal group search in ldap authorization of security realm.
+ */
+public final class GroupToPrincipal {
+
+    private final String baseDn;
+    private final String groupDnAttribute;
+    private final String groupName;
+    private final String groupNameAttribute;
+    private final Boolean iterative;
+    private final Boolean preferOriginalConnection;
+    private final String principalAttribute;
+    private final Boolean recursive;
+    private final String searchBy;
+    private final LdapCache cache;
+
+    private GroupToPrincipal(Builder builder) {
+        this.baseDn = builder.baseDn;
+        this.groupDnAttribute = builder.groupDnAttribute;
+        this.groupName = builder.groupName;
+        this.groupNameAttribute = builder.groupNameAttribute;
+        this.iterative = builder.iterative;
+        this.preferOriginalConnection = builder.preferOriginalConnection;
+        this.principalAttribute = builder.principalAttribute;
+        this.recursive = builder.recursive;
+        this.searchBy = builder.searchBy;
+        this.cache = builder.cache;
+    }
+
+    public String getBaseDn() {
+        return baseDn;
+    }
+
+    public String getGroupDnAttribute() {
+        return groupDnAttribute;
+    }
+
+    public String getGroupName() {
+        return groupName;
+    }
+
+    public String getGroupNameAttribute() {
+        return groupNameAttribute;
+    }
+
+    public Boolean getIterative() {
+        return iterative;
+    }
+
+    public Boolean getPreferOriginalConnection() {
+        return preferOriginalConnection;
+    }
+
+    public String getPrincipalAttribute() {
+        return principalAttribute;
+    }
+
+    public Boolean getRecursive() {
+        return recursive;
+    }
+
+    public String getSearchBy() {
+        return searchBy;
+    }
+
+    public LdapCache getCache() {
+        return cache;
+    }
```

Comment:
Do the methods have to be `public`? I mean, is the class meant to be used for other things than creating a command? If it's only for creating the command, the methods can be omitted and the `private` variables made package-protected.